src/rpdk/core/invoke.py

```diff
@@ -76,6 +76,8 @@ def get_contract_client(args, project):
 def prepare_payload_for_reinvocation(payload, response, artifact_type):
     if artifact_type == ARTIFACT_TYPE_RESOURCE:
         payload["callbackContext"] = response.get("callbackContext")
+        if ("resourceModel" in response) and ("requestData" in payload):
+            payload["requestData"]["resourceProperties"] = response.get("resourceModel")
```

Comment:
Should we default to a previous one of its not returned? None properties might lead to confusing results. 
`payload["requestData"]["resourceProperties"] = response.get("resourceModel") or payload["requestData"]["resourceProperties"]`

Reply:
If it's not returned we wouldn't touch the payload.